tplinkrouterc6u/client.py

```diff
@@ -1428,11 +1428,136 @@ def _prepare_data(self, data: str, is_login: bool) -> tuple[str, str]:
         return signature, encrypted_data
 
 
+class TPLinkXDR3010Client(AbstractRouter):
+    _stok = ''
+
+    def __init__(self, host: str, password: str, username: str = 'admin', logger: Logger = None,
+                 verify_ssl: bool = True, timeout: int = 30) -> None:
+        super().__init__(host, password, username, logger, verify_ssl, timeout)
+        self._session = Session()
+
+    def supports(self) -> bool:
+        response = self._session.get(self.host)
+        return response.text.index('TL-XDR3010') >= 0
```

Comment:
Is there a way to check supports not by checking the model? Adding routers model to the method supports - greatly reduces the possibilities. As I was very surprised how many different types routers have same API

Reply:
Or... try to login in? If it works, that means it's supported.

Reply:
This might be a bit tricky; I used `Session().get()` directly for the request. Now, none of the other routes have been tested with this method.

Reply:
Ok. Could you just post here the output from print(response.text)? and change check to if model starts with string - TL-XDR ?

Reply:
The response is like to this:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
	<title>TL-XDR3010易展版</title>
	<meta name="MobileOptimized" content="240" />
	<meta name="viewport" content="width=device-width, height=device-height, initial-scale=1.0, minimum-scale=0.5, maximum-scale=2.0, user-scalable=yes" />
	<link rel="shortcut Icon" href="../web-static/images/icon.ico" type="image/x-icon" />
	<link rel="stylesheet" href="../web-static/dynaform/class.css">
	<script type="text/javascript" src="../web-static/dynaform/class.js"></script>
	<script type="text/javascript" src="../web-static/dynaform/jtopo.js"></script>
</head>
<body>
	<div id="Error"></div>
	<div id="Confirm"></div>
	<div id="Con"></div>
	<div id="Help"></div>
	<div id="Cover"></div>
	<div id="Login"></div>
	<script type="text/javascript">
		var gBeInCNA = "NO";
		var proName="TL-XDR3010易展版";
		pageOnload();
    </script>
</body>
</html>
```

I'm not sure whether the routing for other XDR models is the same across the board. If it is, I should rename the class to `TPLinkXDRClient`, but I don't have other router models to test this on.

Reply:
Yes - It would be great to rename TPLinkXDRClient. After your good suggestion I have changed the files and now the file structure changed. Could you update your MR with changes from main branch and create tplinkrouterc6u/client/xdr.py and put the client to this file?
Also would be great if you could add methods
- get_ipv4_status https://github.com/AlexandrErohin/TP-Link-Archer-C6U/blob/main/tplinkrouterc6u/client/api_cgi_bin.py#L362C9-L362C24
- get_ipv4_reservations https://github.com/AlexandrErohin/TP-Link-Archer-C6U/blob/main/tplinkrouterc6u/client/api_cgi_bin.py#L380C9-L380C30
- get_ipv4_dhcp_leases https://github.com/AlexandrErohin/TP-Link-Archer-C6U/blob/main/tplinkrouterc6u/client/api_cgi_bin.py#L391C9-L391C29